### ISSUE TITLE
Check scanning logic for redirected or non-public post types

### ISSUE THREAD

**Issue by amberhinds** [equalizedigital/accessibility-checker#1342]
This came up on Highland. Their staff post type is (I thought) no longer a public post type, and going to  any individual post (example: https://highland.edu/staff/steve-stadermann/) redirects to the home page. Yet EDAC is scanning and reporting issues even with the redirect. 

EDAC shows issues that oddly look like they are from the admin.

<img width="1227" height="1200" alt="Image" src="https://github.com/user-attachments/assets/a5745ab4-3823-4b45-a901-d2f1ca0d868b" />

Example HTML that makes me think it's scanning something in the editor. This is flagged for a table missing headers:
```
<table style="display: none"><tbody id="inlineedit"> <tr id="inline-edit" class="inline-edit-row inline-edit-row-post quick-edit-row quick-edit-row-post inline-edit-post" style="display: none"> <td colspan="14" class="colspanchange"> <div class="inline-edit-wrapper" role="region" aria-labelledby="quick-edit-legend"> <fieldset class="inline-edit-col-left"> <legend class="inline-edit-legend" id="quick-edit-legend">Quick Edit</legend> <div class="inline-edit-col"> <label> <span class="title">Title</span> <span class="input-text-wrap"><input type="text" name="post_title" class="ptitle" value=""></span> </label> <label> <span class="title">Slug</span> <span class="input-text-wrap"><input type="text" name="post_name" value="" autocomplete="off" spellcheck="false"></span> </label> <fieldset class="inline-edit-date"> <legend><span class="title">Date</span></legend> <div class="timestamp-wrap"><label><span class="screen-reader-text">Month</span><select class="form-required" name="mm"> <option value="01" data-text="Jan" selected="selected">01-Jan</option> <option value="02" data-text="Feb">02-Feb</option> <option value="03" data-text="Mar">03-Mar</option> <option value="04" data-text="Apr">04-Apr</option> <option value="05" data-text="May">05-May</option> <option value="06" data-text="Jun">06-Jun</option> <option value="07" data-text="Jul">07-Jul</option> <option value="08" data-text="Aug">08-Aug</option> <option value="09" data-text="Sep">09-Sep</option> <option value="10" data-text="Oct">10-Oct</option> <option value="11" data-text="Nov">11-Nov</option> <option value="12" data-text="Dec">12-Dec</option> </select></label> <label><span class="screen-reader-text">Day</span><input type="text" name="jj" value="22" size="2" maxlength="2" autocomplete="off" class="form-required" inputmode="numeric"></label>, <label><span class="screen-reader-text">Year</span><input type="text" name="aa" value="2026" size="4" maxlength="4" autocomplete="off" class="form-required" inputmode="numeric"></label> at <label><span class="screen-reader-text">Hour</span><input type="text" name="hh" value="11" size="2" maxlength="2" autocomplete="off" class="form-required" inputmode="numeric"></label>:<label><span class="screen-reader-text">Minute</span><input type="text" name="mn" value="24" size="2" maxlength="2" autocomplete="off" class="form-required" inputmode="numeric"></label></div><input type="hidden" id="ss" name="ss" value="18"> </fieldset> <br class="clear"> <label class="inline-edit-author"><span class="title">Author</span><select name="post_author" class="authors"> <option value="299">Amy Fairweather (afairweather)</option> <option value="247">Brian Leahy (bleahy)</option> <option value="248">Christie Lewis (clewis)</option> <option value="300">Dan Dick (DDick)</option> <option value="267">Deb Hayenga (dhayenga)</option> <option value="273">Equalize Digital (Equalize_Digital)</option> <option value="256">Jeremy Brandt (jeremy)</option> <option value="252">Jodi Rogers (jrogers)</option> <option value="274">Joe Schneider (jschneider)</option> <option value="265">Kam Merryman (kmerryman)</option> <option value="255">Laura Watson (lwatson)</option> <option value="241">Liz Gerber (Liz)</option> <option value="298">Michael Nguyen (mnguyen)</option> <option value="263">Pat Dunn (pdunn)</option> <option value="244">Patti Potter (Patti)</option> <option value="245">Pete Fink (pfink)</option> <option value="266">Rebecca Anderson (randerson)</option> <option value="272">Stacey Brown (sbrown)</option> <option value="249">Stacey Moore (smoore)</option> <option value="296">Stephanie Hintz (stephanie.hintz)</option> <option value="293">Sydni Lower (SLower)</option> <option value="297">Tabitha Engel (tengel)</option> <option value="246">Terri Grimes (tgrimes)</option> </select></label> <div class="inline-edit-group wp-clearfix"> <label class="alignleft"> <span class="title">Password</span> <span class="input-text-wrap"><input type="text" name="post_password" class="inline-edit-password-input" value=""></span> </label> <span class="alignleft inline-edit-or"> –OR– </span> <label class="alignleft inline-edit-private"> <input type="checkbox" name="keep_private" value="private"> <span class="checkbox-title">Private</span> </label> </div> </div> </fieldset> <fieldset class="inline-edit-col-center inline-edit-categories"> <div class="inline-edit-col"> <span class="title inline-edit-categories-label">Categories</span> <input type="hidden" name="post_category[]" value="0"> <ul class="cat-checklist category-checklist"> <li id="in-category-220-1" class="popular-category"><label class="selectit"><input value="220" type="checkbox" name="post_category[]" id="in-category-220-2"> Staff</label></li> <li id="in-category-1-1" class="popular-category"><label class="selectit"><input value="1" type="checkbox" name="post_category[]" id="in-category-1-2"> Student</label></li> </ul> </div> </fieldset> <fieldset class="inline-edit-col-right"> <div class="inline-edit-col"> <div class="inline-edit-tags-wrap"> <label class="inline-edit-tags"> <span class="title">Tags</span> <textarea data-wp-taxonomy="post_tag" cols="22" rows="1" name="tax_input[post_tag]" class="tax_input_post_tag" aria-describedby="inline-edit-post_tag-desc"></textarea> </label> <p class="howto" id="inline-edit-post_tag-desc">Separate tags with commas</p> </div> <div class="inline-edit-group wp-clearfix"> <label class="alignleft"> <input type="checkbox" name="comment_status" value="open"> <span class="checkbox-title">Allow Comments</span> </label> <label class="alignleft"> <input type="checkbox" name="ping_status" value="open"> <span class="checkbox-title">Allow Pings</span> </label> </div> <div class="inline-edit-group wp-clearfix"> <label class="inline-edit-status alignleft"> <span class="title">Status</span> <select name="_status"> <option value="publish">Published</option> <option value="future">Scheduled</option> <option value="pending">Pending Review</option> <option value="draft">Draft</option> </select> </label> <label class="alignleft"> <input type="checkbox" name="sticky" value="sticky"> <span class="checkbox-title">Make this post sticky</span> </label> </div> </div> </fieldset> <div class="submit inline-edit-save"> <input type="hidden" id="_inline_edit" name="_inline_edit" value="1466f554f1"> <button type="button" class="button button-primary save">Update</button> <button type="button" class="button cancel">Cancel</button> <span class="spinner"></span> <input type="hidden" name="post_view" value="list"> <input type="hidden" name="screen" value="edit-post"> <div class="notice notice-error notice-alt inline hidden"><p class="error"></p></div> </div> </div> <!-- end of .inline-edit-wrapper --> </td></tr> <tr id="bulk-edit" class="inline-edit-row inline-edit-row-post bulk-edit-row bulk-edit-row-post bulk-edit-post" style="display: none"> <td colspan="14" class="colspanchange"> <div class="inline-edit-wrapper" role="region" aria-labelledby="bulk-edit-legend"> <fieldset class="inline-edit-col-left"> <legend class="inline-edit-legend" id="bulk-edit-legend">Bulk Edit</legend> <div class="inline-edit-col"> <div id="bulk-title-div"> <div id="bulk-titles"></div> </div> </div> </fieldset><fieldset class="inline-edit-col-center inline-edit-categories"> <div class="inline-edit-col"> <span class="title inline-edit-categories-label">Categories</span> <input type="hidden" name="post_category[]" value="0"> <ul class="cat-checklist category-checklist"> <li id="bulk-edit-in-category-220-1" class="popular-category"><label class="selectit"><input value="220" type="checkbox" name="post_category[]" id="bulk-edit-in-category-220-2"> Staff</label></li> <li id="bulk-edit-in-category-1-1" class="popular-category"><label class="selectit"><input value="1" type="checkbox" name="post_category[]" id="bulk-edit-in-category-1-2"> Student</label></li> </ul> </div> </fieldset> <fieldset class="inline-edit-col-right"><div class="inline-edit-tags-wrap"> <label class="inline-edit-tags"> <span class="title">Tags</span> <textarea data-wp-taxonomy="post_tag" cols="22" rows="1" name="tax_input[post_tag]" class="tax_input_post_tag" aria-describedby="inline-edit-post_tag-desc"></textarea> </label> <p class="howto" id="inline-edit-post_tag-desc">Separate tags with commas</p> </div> <div class="inline-edit-col"> <label class="inline-edit-author"><span class="title">Author</span><select name="post_author" class="authors"> <option value="-1">— No Change —</option> <option value="299">Amy Fairweather (afairweather)</option> <option value="247">Brian Leahy (bleahy)</option> <option value="248">Christie Lewis (clewis)</option> <option value="300">Dan Dick (DDick)</option> <option value="267">Deb Hayenga (dhayenga)</option> <option value="273">Equalize Digital (Equalize_Digital)</option> <option value="256">Jeremy Brandt (jeremy)</option> <option value="252">Jodi Rogers (jrogers)</option> <option value="274">Joe Schneider (jschneider)</option> <option value="265">Kam Merryman (kmerryman)</option> <option value="255">Laura Watson (lwatson)</option> <option value="241">Liz Gerber (Liz)</option> <option value="298">Michael Nguyen (mnguyen)</option> <option value="263">Pat Dunn (pdunn)</option> <option value="244">Patti Potter (Patti)</option> <option value="245">Pete Fink (pfink)</option> <option value="266">Rebecca Anderson (randerson)</option> <option value="272">Stacey Brown (sbrown)</option> <option value="249">Stacey Moore (smoore)</option> <option value="296">Stephanie Hintz (stephanie.hintz)</option> <option value="293">Sydni Lower (SLower)</option> <option value="297">Tabitha Engel (tengel)</option> <option value="246">Terri Grimes (tgrimes)</option> </select></label> <div class="inline-edit-group wp-clearfix"> <label class="alignleft"> <span class="title">Comments</span> <select name="comment_status"> <option value="">— No Change —</option> <option value="open">Allow</option> <option value="closed">Do not allow</option> </select> </label> <label class="alignright"> <span class="title">Pings</span> <select name="ping_status"> <option value="">— No Change —</option> <option value="open">Allow</option> <option value="closed">Do not allow</option> </select> </label> </div> <div class="inline-edit-group wp-clearfix"> <label class="inline-edit-status alignleft"> <span class="title">Status</span> <select name="_status"> <option value="-1">— No Change —</option> <option value="publish">Published</option> <option value="private">Private</option> <option value="pending">Pending Review</option> <option value="draft">Draft</option> </select> </label> <label class="alignright"> <span class="title">Sticky</span> <select name="sticky"> <option value="-1">— No Change —</option> <option value="sticky">Sticky</option> <option value="unsticky">Not Sticky</option> </select> </label> </div> </div> </fieldset> <div class="submit inline-edit-save"> <input type="submit" name="bulk_edit" id="bulk_edit" class="button button-primary" value="Update"> <button type="button" class="button cancel">Cancel</button> <input type="hidden" name="post_view" value="list"> <input type="hidden" name="screen" value="edit-post"> <div class="notice notice-error notice-alt inline hidden"><p class="error"></p></div> </div> </div> <!-- end of .inline-edit-wrapper --> </td></tr> </tbody></table>
```

Highland's post types are registered in a super weird way. I actually think that while I thought this post type was no longer public, it actually still is? [Here's Highland's registration function.](https://github.com/equalizedigital/highland-edu-theme/blob/6f4828af8fa49c16a62fe74a6549676abc0c3c49/includes/cpt_functions.php#L36)

I don't see a view link here:

<img width="536" height="283" alt="Image" src="https://github.com/user-attachments/assets/393e10e8-f029-401a-babf-0cfb023c1cc4" />

However there are view links when you're editing an individual post:

<img width="1710" height="474" alt="Image" src="https://github.com/user-attachments/assets/a1b61e07-81b9-4bfa-9d29-bb09bdca1601" />

In Accessibility Checker Settings, Staff is a post type that is available to be scanned.

<img width="784" height="850" alt="Image" src="https://github.com/user-attachments/assets/6a24a94f-ed89-4e0b-8a7d-c387f1992c00" />

I have no idea what is redirecting the posts on Highland. This may be a bug specific to that website and an edgecase do to its odd theme code, but it's worth investigating a few things:

1. If the post type is, indeed, not public, why is it available in the list of post types on the settings page? Does this list filter out nonpublic post types?
2. Why is EDAC returning scan results on these posts rather than aborting with the redirect and why do the scan results appear to be of the admin rather than the homepage (where the redirect goes?

**Comment by SteveJonesDev** [comment]
**Finding: EDAC scans the preview URL, not the public permalink**

After digging into the scanning logic, the root cause is clearer.

For all posts (except the front page/blog index), EDAC constructs the scan URL using `get_preview_post_link()` rather than `get_permalink()`. See `admin/class-enqueue-admin.php`. This is intentional — it allows scanning drafts and private content before they're publicly accessible.

For Highland's staff CPT, the **public** URL (`highland.edu/staff/steve-stadermann/`) redirects to the homepage. But the **preview URL** (`?p=123&preview=true`) does not redirect — it resolves fine. So EDAC never sees the redirect and proceeds with the scan.

**Redirect detection exists but doesn't apply here.** `checkPage.js` checks whether `edac_pageScanner` is still present in the iframe's final URL after load. If the preview URL itself redirected away, this would catch it and abort the scan. But since the preview URL doesn't redirect, the check never fires.

**The admin HTML is the strange part.** The Quick Edit / Bulk Edit table HTML is exclusive to WP admin list tables (`edit.php`) — it won't appear on a frontend preview or even a redirected homepage. This suggests that for this particular CPT configuration, WordPress's preview mechanism may be routing the `?p=123&preview=true` request to the admin posts list rather than a frontend render. Worth reproducing in a staging environment to confirm.

**On the settings page question:** The staff CPT is likely still registered as `public: true` (Highland's theme code does something unusual but may not actually set `public: false`). Whatever causes the frontend redirect is happening at a different layer. The settings page shows it because it passes the public check.

**Next steps:**
- Confirm whether the preview URL for a Highland staff post actually resolves to `edit.php` (browser devtools / network tab while scanning)
- Determine whether the fix should be: (a) detecting when a preview URL lands on an admin page and aborting, (b) falling back to public permalink when preview URL misbehaves, or (c) a pre-scan check that validates the CPT is actually publicly queryable before scanning
